src/App.tsx

```diff
@@ -1,26 +1,54 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from './app/hooks';
+import { useCallback, useEffect, useState } from 'react';
+import { getTodos } from './api';
+import { setTodos } from './features/todos';
+import { selectCurrentTodo, selectTodos } from './features/todoSelectors';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App: React.FC = () => {
+  const dispatch = useDispatch();
+  const todos = useAppSelector(selectTodos);
+  const currentTodo = useAppSelector(selectCurrentTodo);
+  const [isLoading, setIsLoading] = useState(false);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  const fetchTodos = useCallback(async () => {
+    setIsLoading(true);
+    try {
+      const fetchedTodos = await getTodos();
+
+      dispatch(setTodos(fetchedTodos));
+    } finally {
+      setIsLoading(false);
+    }
+  }, [dispatch]);
+
+  useEffect(() => {
+    fetchTodos();
+  }, [fetchTodos]);
+
+  return (
+    <>
+      <div className="section">
+        <div className="container">
+          <div className="box">
+            <h1 className="title">Todos:</h1>
+
+            <div className="block">
+              <TodoFilter />
+            </div>
 
-          <div className="block">
-            <Loader />
-            <TodoList />
+            <div className="block">
+              {isLoading && <Loader />}
+              {!isLoading && !!todos.length && <TodoList />}
```

Comment:
The conditional rendering logic for `TodoList` might be slightly misleading. The condition `!!todos.length` will render `TodoList` even if `todos` is an empty array, as `!!0` evaluates to `false`. Consider changing the condition to `todos.length > 0` to ensure `TodoList` is only rendered when there are todos available.